qemu/tests/virtio_serial_throttling.py

```diff
@@ -0,0 +1,53 @@
+from avocado.utils import process
+from virttest import error_context
+
+from qemu.tests.virtio_serial_file_transfer import transfer_data
+
+import socket
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Induce throttling by opening chardev but not reading data from it:
+    1) Start guest with virtio-serial with unix option
+    2) On the host, open the chardev but don't read from it
+    3) In the guest, write to the virtio-serial port.
+    4) repeat step 1 to 3 with tcp option.
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_chardev_params(vm, serial_id):
+        serial_device = vm.devices.get(serial_id)
+        chardev_id = serial_device.get_param('chardev')
+        chardev_device = vm.devices.get(chardev_id)
+        chardev_params = chardev_device.params
+        return chardev_params
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    serial_id = params.objects('serials')[-1]
+    chardev_path = get_chardev_params(vm, serial_id)['path']
+    session = vm.wait_for_login()
+    sock = socket.socket(socket.AF_UNIX)
+    sock.connect(chardev_path)
+    a = session.cmd_output('dd if=/dev/zero of=/dev/vport0p2')
+    transfer_data(params, vm, sender='host')
```

Comment:
This is not what the case wants to do, you should send data only from the guest but not receive data on the host.
please use lonely cmd to send data not the whole function.

Reply:
Hi but from polarion it said "4.repeat step 2 to step 3 when transfer data from host to guest]"
And in transfer_data function the sender is host only and not receive data on host.
If it receive on host it should be  transfer_data(params, vm, sender='both') or sender='guest'

Reply:
> Hi but from polarion it said "4.repeat step 2 to step 3 when transfer data from host to guest]"
> And in transfer_data function the sender is host only and not receive data on host.
> If it receive on host it should be transfer_data(params, vm, sender='both') or sender='guest'

the parameter `sender` means that who send data in a complete process of transferring data, the function `transfer_data` is a complete follow that will finish data transfer, include sending data and receiving data, but as for this case, the data transfer is not complete so we can't use this function `transfer_data`

Reply:
@michiboo I just noticed our test framework support this.  The virtio ports in `vm.virtio_ports`  is an instance of `qemu_virtio_port.VirtioSerial` (or `qemu_virtio_port.VirtioConsole`), which support `open`, `close` host socket, and 'self.sock' that can be used to send data. we do not need to write it again in this case.
You can refer `virtio_serial_empty_line.py`